crates/movement/src/pathing.rs

```diff
@@ -67,6 +83,9 @@ fn follow_path(
     objects
         .par_iter_mut()
         .for_each_mut(|(transform, mut path, mut movement)| {
+            if movement.paused() {
```

Comment:
Instead of having a separate system `check_battery()` I would change this condition to:

```rust
if battery.energy() <= 0. {
    movement.stop();
    return;
}
```